File: Java/AppConcesionario/app/src/main/java/com/dam/aplicacin_temtica_libre/MainActivity.java
package com.dam.aplicacin_temtica_libre;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    private EditText etUsuario, et_contraseñaCrear; // EditText para el nombre de usuario y la contraseña

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Habilitar el modo de borde a borde
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main); // Establecer el diseño de la actividad

        // Configurar el comportamiento de los insets de ventana
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom); // Ajustar los paddings
            return insets;
        });

        // Obtener referencias a los EditText
        etUsuario = findViewById(R.id.et_usuario); // Campo de texto para el nombre de usuario
        et_contraseñaCrear = findViewById(R.id.et_contraseñaCrear); // Campo de texto para la contraseña
    }

    // Método para iniciar la actividad de creación de cuenta
    public void Crear_Cuenta(View view) {
        Intent i = new Intent(this, Crear_cuenta.class); // Crear un nuevo Intent para la actividad de crear cuenta
        startActivity(i); // Iniciar la actividad
    }

    // Método para iniciar sesión
    public void iniciarSesion(View view) {
        // Obtener el texto del EditText y eliminar espacios en blanco al principio y al final
        String usuario = etUsuario.getText().toString().trim(); // Obtener el nombre de usuario
        String contraseña = et_contraseñaCrear.getText().toString().trim(); // Obtener la contraseña

        // Obtener SharedPreferences para almacenar datos de usuarios
        SharedPreferences preferencias = getSharedPreferences("usuarios", Context.MODE_PRIVATE);

        // Recuperar los datos almacenados en SharedPreferences
        String storedUsuario = preferencias.getString("usuario_" + usuario, null); // Obtener el usuario almacenado
        String storedContraseña = preferencias.getString("contraseña_" + usuario, null); // Obtener la contraseña almacenada

        // Verificar si el usuario y la contraseña coinciden
        if (storedUsuario != null && storedContraseña != null) {
            if (contraseña.equals(storedContraseña)) {
                // Si la contraseña es correcta, iniciar la actividad de inicio
                Intent intent = new Intent(this, Inicia.class);
                intent.putExtra("usuario", usuario); // Pasar el nombre de usuario a la siguiente actividad
                startActivity(intent); // Iniciar la actividad
                finish(); // Finalizar esta actividad
                Toast.makeText(this, "Inicio de sesión exitoso", Toast.LENGTH_SHORT).show(); // Mensaje de éxito

            } else {
                // Si la contraseña es incorrecta, mostrar un mensaje
                Toast.makeText(this, "Contraseña incorrecta", Toast.LENGTH_SHORT).show();
            }
        } else {
            // Si el usuario no se encuentra, mostrar un mensaje
            Toast.makeText(this, "Usuario no encontrado", Toast.LENGTH_SHORT).show();
        }
    }
}

File: Kotlin/proyecto_viviendas_Slide/PROYECTO_RECUPERACION_VIVIENDAS/app/src/main/java/com/example/proyecto_recuperacion_viviendas/MainActivity.kt
package com.example.proyecto_recuperacion_viviendas

import android.content.Intent
import android.os.Build
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.ImageView
import androidx.annotation.RequiresApi
import com.bumptech.glide.Glide

class MainActivity : AppCompatActivity() {
    @RequiresApi(Build.VERSION_CODES.R)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val btn_acceso = findViewById<Button>(R.id.boton_acceso)
        val nombre = findViewById<EditText>(R.id.nombre_acceso)
        val apellido = findViewById<EditText>(R.id.apellido_acceso)


        btn_acceso.setOnClickListener{
            val pantalla_inicio = Intent(this, inicio::class.java)

            if ((apellido.text.toString() != "") and (nombre.text.toString() != "")){

                val pasar_inicio = Intent(this,inicio::class.java)
                pasar_inicio.putExtra("apellido", apellido.text.toString())
                pasar_inicio.putExtra("nombre", nombre.text.toString())
                startActivity(pasar_inicio)

            }
        }

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
            if (resources.configuration.isNightModeActive) {
                window.setBackgroundDrawableResource(R.drawable.background_dark)
            } else {
                window.setBackgroundDrawableResource(R.drawable.background_light)
            }
        } else {
            window.setBackgroundDrawableResource(R.drawable.background_light)
        }

    }
}